### PR TITLE
Custom Tailored Mat-Table (with affected table pages)

### PR DESCRIPTION
**Pull request by albannurkollari**
- [x] Create Table Component from scratch
- [x] Import/Export it in SharedModule so that it can be accessed by any other component/module in the app
- [x] Generate the table design
- [x] Convert the static column blocks of HTML into a `ngFor` directive loop
- [x] Assign specific content in columns, e.g. insert links instead of just strings on Organization Name(s) and Project Name(s)
- [x] Style the table to reflect the same design in every page it is inserted as a child
- [x] Add behavior like hover on rows, header cells etc
- [x] Add Sorting and fix sorting visual behavior
- [ ] Add Pagination
- [x] Add Filtering

P.s. There is a huge contribute from previous developers into helping me achieve this.

Related issue: #223

### CODE REVIEW

**Comment by albannurkollari** [comment]
The number of commits IS irelevant due to rebase conflicts occured during the process of it. Commits I've made are around no more than 30-35.

File: src/app/admin/admin.component.html [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <div class='wrapper'>
     <app-sidebar class='trans-15s'></app-sidebar>
+    <div class='overlay' (mouseup)='triggerToggler($event)'></div>
```
Comment (alayham) [inline src/app/admin/admin.component.html]: is this related to the table component?

File: src/app/admin/admin.component.scss [hunk 2]
```diff
@@ -33,8 +33,9 @@ $sidebarClosedVal: '0';
         overflow: hidden;
     }
 
-    .wrapper::before {
-        content: "";
+    .wrapper::before,
```
Comment (alayham) [inline src/app/admin/admin.component.scss]: Is this related to the table component

File: src/app/admin/admin.component.ts [hunk 3]
```diff
@@ -6,9 +6,16 @@ import { Router } from '@angular/router';
   templateUrl: './admin.component.html',
   styleUrls: ['./admin.component.scss']
 })
+
 export class AdminComponent implements OnInit {
+  triggerToggler: Function = (evt): void => {
+    const toggler = document.querySelectorAll('.sidebar-toggle.expanded')[0];
```
Comment (alayham) [inline src/app/admin/admin.component.ts]: Is this related to the table component?

File: src/app/admin/organization-list/organization-list.component.ts [hunk 4]
```diff
@@ -1,115 +1,82 @@
...
-import { DataSource } from '@angular/cdk/table';
-import { MatPaginator, MatSort } from '@angular/material';
-import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+import { Component, OnInit } from '@angular/core';
 import { Observable } from 'rxjs/Observable';
-import 'rxjs/add/observable/of';
-import 'rxjs/add/operator/startWith';
-import 'rxjs/add/observable/merge';
-import 'rxjs/add/operator/map';
+import '../../shared/rxjs.operators';
```
Comment (alayham) [inline src/app/admin/organization-list/organization-list.component.ts]: Is this related to the table component

File: src/app/shared/rxjs.operators.ts [hunk 5]
```diff
@@ -10,3 +10,4 @@ import 'rxjs/add/operator/mergeMap';
 import 'rxjs/add/operator/filter';
 import 'rxjs/add/operator/debounceTime';
 import 'rxjs/add/operator/distinctUntilChanged';
+import 'rxjs/add/operator/startWith';
```
Comment (alayham) [inline src/app/shared/rxjs.operators.ts]: Is this related to the table component?
